### ISSUE TITLE
improve java 17 build support

### ISSUE THREAD

**Issue by landryb** [geosolutions-it/MapStore2#9850]
## Description
as of now mapstore 2 doesnt build with java 17.

a first step is to use a more recent version of the `maven-war` plugin, using 3.4.0 instead of 2.1.1.

this allows a build with `-DskipTests` to succeed.

the tests themselves fail to build with the same failure:
```
java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.creation.jmock.ClassImposterizer$3
        at org.mockito.internal.creation.jmock.ClassImposterizer.createProxyClass(ClassImposterizer.java:85)
        at org.mockito.internal.creation.jmock.ClassImposterizer.imposterise(ClassImposterizer.java:62)
        at org.mockito.internal.creation.jmock.ClassImposterizer.imposterise(ClassImposterizer.java:56)
        at org.mockito.internal.creation.CglibMockMaker.createMock(CglibMockMaker.java:23)
        at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:26)
        at org.mockito.internal.MockitoCore.mock(MockitoCore.java:51)
        at org.mockito.Mockito.mock(Mockito.java:1243)
        at org.mockito.Mockito.mock(Mockito.java:1120)
        at it.geosolutions.mapstore.LoadAssetControllerTest.testLoadPngAsset(LoadAssetControllerTest.java:152)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
```
this has probably something to do with an update of the mockito framework ?

**Comment by landryb** [comment]
cf #6935 for a requirements update if java 17 is deemed acceptable. i know mapstore runs fine with java17, and #9851 might allow to build with java17.

**Comment by landryb** [comment]
fixed by #9851
